87.md

```diff
@@ -0,0 +1,129 @@
+NIP-87
+======
+
+Ecash Mint Discoverability
+--------------------------------
+
+`draft` `optional`
+
+This NIP describes `kind:38173` and `kind:18173`: a way to discover ecash mints, their capabilities, and people who recommend them.
+
+## Rationale
+
+Nostr's discoverability and transparent event interaction is one of its most interesting/novel mechanics.
+This NIP provides a simple way for users to discover ecash mints recommended by other users and to interact with them.
+
+### Parties involved
+
+There are three actors to this workflow:
+
+* An ecash mint operator, announces their mint and its capabilities.
+    * Publishes `kind:38173`, detailing how to connect to it and its capabilities.
+* user A, who recommends an app that handles a specific event kind
+    * Publishes `kind:18173`
+* user B, who seeks a recommendation for an ecash mint
+    * Queries for `kind:18173` and, based on results, queries for `kind:38173`
+
+## Events
+
+### Recommendation event
+```json
+{
+  "kind": 18173,
+  "pubkey": <recommender-user-pubkey>,
+  "tags": [
+    ["u", <recommended-cashu-mint-url>, "cashu"],
+    ["u", <recommended-fedimint-invite-code>, "fedimint"],
+    ["u", <recommended-cashu-mint-url>, "cashu"],
+    ["a", "38173:mint1-pubkey:<d-identifier>", "wss://relay1", "cashu"],
+    ["a", "38173:mint2-pubkey:<d-identifier>", "wss://relay2", "fedimint"]
+  ]
+}
+```
+
+`u` tags are used to point directly to the ecash mint.
+The first value of the tag is the URL or invite code of the ecash mint, and the second value is the type of ecash mint (currently `cashu` or `fedimint`).
+Multiple `u` tags can appear on the same `kind:18173`.
+
+`a` tags are used to point to the `kind:38173` event of the ecash mint.
+The first value of the tag is the `kind:38173` event identifier, the second value of the tag is a relay hint, and the third value of the tag is the type of ecash mint (currently `cashu` or `fedimint`).
+
+## Ecash Mint Information
+
+Cashu mints SHOULD publish `kind:38173` events to announce their capabilities and how to connect to them.
+
+For cashu mints, the `u` tag SHOULD be the URL to the cashu mint and it should list the nuts that the cashu mint supports.
+
+```json
+{
+  "kind": 38173,
+  "pubkey": "<application-pubkey>",
+  "content": "<optional-kind:0-style-metadata>",
+  "tags": [
+    ["d", <random-id>],
+    ["u", "https://cashu.example.com", "cashu"],
+    ["nuts", "1,2,3,4,5,6,7"]
+  ]
+}
+```
+
+Fedimints SHOULD publish `kind:38173` events to announce their capabilities and how to connect to them.
+
+For fedimints, it should list all known fedimint invite codes in `u` tags and it should list the modules it supports.
+
+```json
+{
+  "kind": 38173,
+  "pubkey": "<application-pubkey>",
+  "content": "<optional-kind:0-style-metadata>",
+  "tags": [
+    ["d", <random-id>],
+    ["u", "fed11abc..", "fedimint"],
+    ["u", "fed11xyz..", "fedimint"],
+    ["modules", "lightning,wallet,mint"]
+  ]
+}
```

Comment:
Should the federation ID be referenced and published here in some way?

Federations invite codes can change and there could be many. It is going to be very difficult to aggregate and parse them all when they could be referencing the same federation. The federation ID is what the user is "joining" not just a single invite code.